lib/AST/ASTImporter.cpp

```diff
@@ -908,8 +908,27 @@ bool ASTNodeImporter::ImportDeclParts(NamedDecl *D, DeclContext *&DC,
                                       DeclarationName &Name, 
                                       NamedDecl *&ToD,
                                       SourceLocation &Loc) {
+  // Check if RecordDecl is in FunctionDecl parameters to avoid infinite loop.
```

Comment:
insert here the example code that is handled by this part

Reply:
Done